### ISSUE TITLE
fix PodGroup being incorrectly deleted due to frequent creation and deletion of pods

### ISSUE THREAD

**Issue by guoqinwill** [volcano-sh/volcano#3374]
**What happened**:
The workload frequently changes its replica count from 1 to 0 to 1 after running，its pods occasionally remain in a pending state indefinitely.

**What you expected to happen**:
During the operation, all pods are running normally.

**How to reproduce it (as minimally and precisely as possible)**:
1、Create a deployment workload.
2、Frequently modify its replica count, changing from 1 to 0 and then back to 1.
3、Observe the status of its pods.

**Anything else we need to know?**:

**Environment**:
- Volcano Version: 1.9.0
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**This problem occurs because**:
![image](https://github.com/volcano-sh/volcano/assets/151122323/442ed356-daef-46c6-9a25-a16d51b3743c)
When we perform the above operations frequently on the replica count of the workload, the state of the podgroup in the environment changes as shown in the "before fix" section of the diagram: the podgroup will attempt to delete the pg when it detects that the pod count is 0. However, there may be delays caused by actions such as resource release during the deletion process. If the deletion fails, it will retry and add the pg back to the delayed deletion queue, which will execute the add action and perform deletion after a certain time interval. If, at this point, we again set the pod count to 1, and this operation occurs before the "retry" adds the pg back, it will result in the pg being created and immediately deleted, causing the newly created pods to be unable to find the new pg and remain in the "pending" state. The error log is as follows:

```
I0320 14:26:31.045519       7 util.go:56] schedulerPodName  is responsible to Pod default/gqtest-3-54d4c4bd54-zwwl7
I0320 14:26:31.045550       7 event_handlers.go:388] Added pod <default/gqtest-3-54d4c4bd54-zwwl7> into cache.
I0320 14:26:31.055090       7 util.go:94] schedulerPodName  is responsible to PodGroup default/podgroup-26ffb840-9e46-4be5-9c64-a5fcd227f8fb
I0320 14:26:31.055110       7 event_handlers.go:731] Add PodGroup(podgroup-26ffb840-9e46-4be5-9c64-a5fcd227f8fb) into cache, spec(v1beta1.PodGroupSpec{MinMember:1, MinTaskMember:map[string]int32(nil), Queue:"default", PriorityClassName:"", MinResources:(*v1.ResourceList)(0xc00106e078)})
I0320 14:26:31.066539       7 util.go:56] schedulerPodName  is responsible to Pod default/gqtest-3-54d4c4bd54-zwwl7
I0320 14:26:31.066557       7 util.go:56] schedulerPodName  is responsible to Pod default/gqtest-3-54d4c4bd54-zwwl7
I0320 14:26:31.066577       7 event_handlers.go:413] Updated pod <default/gqtest-3-54d4c4bd54-zwwl7> in cache.
I0320 14:26:31.711289       7 cache.go:1016] Job <default/podgroup-26ffb840-9e46-4be5-9c64-a5fcd227f8fb:default/podgroup-26ffb840-9e46-4be5-9c64-a5fcd227f8fb> was deleted.
I0320 14:26:32.005468       7 scheduler.go:102] Start scheduling ...
I0320 14:26:32.005532       7 node_info.go:319] set the node 192.168.88.34 status to Ready.
I0320 14:26:32.005575       7 node_info.go:217] imageStates is map[]
I0320 14:26:32.005602       7 node_info.go:319] set the node 192.168.66.134 status to Ready.
I0320 14:26:32.005624       7 node_info.go:217] imageStates is map[]
I0320 14:26:32.005643       7 node_info.go:319] set the node 192.168.116.5 status to Ready.
I0320 14:26:32.005662       7 node_info.go:217] imageStates is map[]
I0320 14:26:32.005689       7 cache.go:1242] The priority of job <kube-system/podgroup-5695fc91-30a9-4de0-9a9c-5bfb7b6c9ffb> is <system-cluster-critical/2000000000>
I0320 14:26:32.005705       7 cache.go:1242] The priority of job <kube-system/podgroup-f9ed94ab-0a6f-4b06-b42a-1486f592ed03> is <system-node-critical/2000001000>
I0320 14:26:32.005706       7 cache.go:1242] The priority of job <kube-system/podgroup-5dd1d761-6c34-41a4-9c7f-0754c7b8b320> is </0>
I0320 14:26:32.005723       7 cache.go:1242] The priority of job <default/podgroup-b1aebf55-4240-40bc-b161-693341e1b556> is </0>
I0320 14:26:32.005736       7 cache.go:1242] The priority of job <default/podgroup-d87643bf-f8ae-4ed1-91b3-c8cb710e27ea> is </0>
I0320 14:26:32.005716       7 cache.go:1242] The priority of job <monitoring/podgroup-0f869aea-909a-4f0f-9cba-4dead7bec834> is <system-node-critical/2000001000>
I0320 14:26:32.005744       7 cache.go:1242] The priority of job <kube-system/podgroup-15cf84e0-1102-490f-a4c7-f6ddf53ef495> is </0>
I0320 14:26:32.005756       7 cache.go:1242] The priority of job <monitoring/podgroup-c8df845d-34eb-4aa3-b900-dd1863ffaa29> is </0>
I0320 14:26:32.005719       7 cache.go:1242] The priority of job <default/podgroup-0893f71d-c803-479f-9cb2-6c4e3465801a> is </0>
I0320 14:26:32.005764       7 cache.go:1242] The priority of job <monitoring/podgroup-0f28e2dd-b7b0-4a98-9ec9-4cb5980ea759> is </0>
I0320 14:26:32.005773       7 cache.go:1242] The priority of job <monitoring/podgroup-5626fd74-a4dc-4b59-bdea-a094a929a7a1> is </0>
I0320 14:26:32.005775       7 cache.go:1242] The priority of job <kube-system/podgroup-440c7a51-e4c0-4b36-b615-30d408ab824b> is <system-cluster-critical/2000000000>
I0320 14:26:32.005783       7 cache.go:1242] The priority of job <kube-system/podgroup-558b1eec-5d88-49c9-b521-fcfd5af3b28b> is <system-cluster-critical/2000000000>
I0320 14:26:32.005805       7 cache.go:1278] There are <13> Jobs, <1> Queues and <3> Nodes in total for scheduling.
I0320 14:26:32.005823       7 session.go:186] Open Session 9e71a305-6df9-4575-a8ee-5362f36c117c with <13> Job and <1> Queues
I0320 14:26:32.006304       7 drf.go:191] Total Allocatable cpu 9770.00, memory 36117250048.00, attachable-volumes-csi-proxy.csi.everest.io 6442450941.00, hugepages-1Gi 0.00, attachable-volumes-csi-nas.csi.everest.io 6442450941.00, attachable-volumes-csi-local.csi.everest.io 6442450941.00, pods 96.00, attachable-volumes-csi-sfsturbo.csi.everest.io 6442450941.00, ephemeral-storage 283729681146000.00, attachable-volumes-csi-obs.csi.everest.io 6442450941.00, hugepages-2Mi 0.00, attachable-volumes-csi-disk.csi.everest.io 162.00
I0320 14:26:32.006410       7 binpack.go:164] Enter binpack plugin ...
```

after fix:
The above problem arises because when deleting the pg, it cannot differentiate between the newly added pg that needs to participate in scheduling and the old pg being deleted due to retry. It solely relies on the pg name for identification, lacking a unique identifier, which may result in the new pg being deleted immediately after being added. To address this issue, we have made a fix by distinguishing between new and old pgs. Referring to the logs below, when the pod count changes from 0 to 1, a new pg with the same name but a different UID is created. For instance, for "podgroup-b1aebf55-4240-40bc-b161-693341e1b556", the "oldpgversion" and "newpgversion" respectively represent the newly created PgUID in the cache and the PgUID from the previous retry action attempting to delete again:
```
I0328 14:51:19.825704       7 util.go:56] schedulerPodName  is responsible to Pod default/gqtest-2-b8fd74666-vnxfl
I0328 14:51:19.825734       7 event_handlers.go:388] Added pod <default/gqtest-2-b8fd74666-vnxfl> into cache.
I0328 14:51:19.931704       7 util.go:94] schedulerPodName  is responsible to PodGroup default/podgroup-b1aebf55-4240-40bc-b161-693341e1b556
I0328 14:51:19.931727       7 event_handlers.go:731] Add PodGroup(podgroup-b1aebf55-4240-40bc-b161-693341e1b556) into cache, spec(v1beta1.PodGroupSpec{MinMember:1, MinTaskMember:map[string]int32(nil), Queue:"default", PriorityClassName:"", MinResources:(*v1.ResourceList)(0xc000cfc548)})
I0328 14:51:19.946585       7 util.go:56] schedulerPodName  is responsible to Pod default/gqtest-2-b8fd74666-vnxfl
I0328 14:51:19.946603       7 util.go:56] schedulerPodName  is responsible to Pod default/gqtest-2-b8fd74666-vnxfl
I0328 14:51:19.946619       7 event_handlers.go:413] Updated pod <default/gqtest-2-b8fd74666-vnxfl> in cache.
I0328 14:51:20.328739       7 leaderelection.go:285] successfully renewed lease kube-system/volcano
I0328 14:51:20.472003       7 cache.go:1036] oldpgversion:70915324-4b00-416a-99cb-ec4c5cf27b2e, newpgversion:201e5032-6fe0-4f5e-93e9-b4c43c749c08
I0328 14:51:20.472043       7 cache.go:1044] job default/podgroup-b1aebf55-4240-40bc-b161-693341e1b556 resourceversion 70915324-4b00-416a-99cb-ec4c5cf27b2e
I0328 14:51:20.720671       7 scheduler.go:102] Start scheduling ...
I0328 14:51:20.720739       7 node_info.go:319] set the node 192.168.88.34 status to Ready.
I0328 14:51:20.720763       7 node_info.go:217] imageStates is map[]
I0328 14:51:20.720789       7 node_info.go:319] set the node 192.168.116.5 status to Ready.
I0328 14:51:20.720801       7 node_info.go:217] imageStates is map[]
I0328 14:51:20.720821       7 node_info.go:319] set the node 192.168.66.134 status to Ready.
I0328 14:51:20.720838       7 node_info.go:217] imageStates is map[]
I0328 14:51:20.720857       7 cache.go:1271] The priority of job <kube-system/podgroup-07f82cbe-870a-4847-afd7-a86299792f56> is <system-cluster-critical/2000000000>
I0328 14:51:20.720873       7 cache.go:1271] The priority of job <kube-system/podgroup-440c7a51-e4c0-4b36-b615-30d408ab824b> is <system-cluster-critical/2000000000>
I0328 14:51:20.720883       7 cache.go:1271] The priority of job <kube-system/podgroup-558b1eec-5d88-49c9-b521-fcfd5af3b28b> is <system-cluster-critical/2000000000>
I0328 14:51:20.720894       7 cache.go:1271] The priority of job <kube-system/podgroup-f9ed94ab-0a6f-4b06-b42a-1486f592ed03> is <system-node-critical/2000001000>
I0328 14:51:20.720907       7 cache.go:1271] The priority of job <monitoring/podgroup-224d85ac-25ab-41d0-9e39-5917cb421c75> is </0>
I0328 14:51:20.720915       7 cache.go:1271] The priority of job <default/podgroup-0893f71d-c803-479f-9cb2-6c4e3465801a> is </0>
I0328 14:51:20.720942       7 cache.go:1271] The priority of job <default/podgroup-26ffb840-9e46-4be5-9c64-a5fcd227f8fb> is </0>
I0328 14:51:20.720954       7 cache.go:1271] The priority of job <default/podgroup-b1aebf55-4240-40bc-b161-693341e1b556> is </0>
```

**Comment by william-wang** [comment]
@guoqinwill what kind of workload are you using?

**Comment by guoqinwill** [comment]
> @guoqinwill what kind of workload are you using?

Deployment

**Comment by lowang-bh** [comment]
/close 

fixed by #3375
